### ISSUE TITLE
Cannot run ncu as a script in an ESM project

### ISSUE THREAD

**Issue by arklanq** [raineorshine/npm-check-updates#1322]
- [X] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [X] I am using the latest version of `npm-check-updates`
- [X] I am using `node >= 14.14`

Node.js = v18.17.0
Package manager Yarn = 1.22.19
npm-check-updates = 16.13.2

---

## Steps to Reproduce

1. Create package.json with following content:

```
{
  "type": "module",
  "scripts" {
    {
      "check-updates": "ncu",
      "force-updates": "ncu -u"
    }
  }
}
```

> In fact, the script is a bit longer, I shortened it for the sake of code clarity.

2. Run `yarn check-updates`.

3. Observe error:

```
/Users/user-x/path-to-project/node_modules/cli-table3/src/utils.js:1
const stringWidth = require('string-width');
                    ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/user-x/path-to-project/node_modules/string-width/index.js from /Users/user-x/path-to-project/node_modules/cli-table3/src/utils.js not supported.
Instead change the require of index.js in /Users/user-x/path-to-project/node_modules/cli-table3/src/utils.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/Users/user-x/path-to-project/node_modules/cli-table3/src/utils.js:1:21)
    at Object.<anonymous> (/Users/user-x/path-to-project/node_modules/cli-table3/src/table.js:2:15)
    at Object.<anonymous> (/Users/user-x/path-to-project/node_modules/cli-table3/index.js:1:18)
    at Object.<anonymous> (/Users/user-x/path-to-project/node_modules/npm-check-updates/build/src/lib/logging.js:10:38)
    at Object.<anonymous> (/Users/user-x/path-to-project/node_modules/npm-check-updates/build/src/lib/cache.js:10:19)
    at Object.<anonymous> (/Users/user-x/path-to-project/node_modules/npm-check-updates/build/src/cli-options.js:9:17)
    at Object.<anonymous> (/Users/user-x/path-to-project/node_modules/npm-check-updates/build/src/bin/cli.js:35:36) {
  code: 'ERR_REQUIRE_ESM'
}
```

## Current Behavior

Error raised?

## Expected Behavior

NCU should work both in ESM (type: module) context as well as in CJS (type: commonjs) context.

## SIde nodes

Take as an example Next.js technology, their CLI lib "next" works well when defined as a package.json script, no matter the context (ESM/CJS).

**Comment by raineorshine** [comment]
Hi, thanks for reporting.

I have tested on the exact node, yarn, and npm-check-updates you listed and I am not able to reproduce. It runs and prints relevant updates as expected. The test suite is currently passing on node v14, v16, and v18 on both Linux and Windows, so it is unlikely that the bare `ncu` is broken.

It seems like it there is something specific to your installation that is causing the module error.

**Comment by arklanq** [comment]
Holy shit, unimaginable. I tried removing `type: "module"` as suggested, but it still didn't work. So next I have run:

```
rm -rf node_modules yarn.lock && yarn install
```

And guess what... error disappeared... 

From this point on I also cannot reproduce the issue, trying with or without `type: "module"`, reinstalling multiple times node_modules, and still nothing. The error just magically disappeared.

So I have reviewed the changes in yarn.lock file compared to the previous commit. Among a few other changes that look like patch upgrades for packages released meanwhile, the only change that looks suspicious to me is:

![image](https://github.com/raineorshine/npm-check-updates/assets/23725071/ff872849-12fe-45ff-bbe5-73736d5a2f65)

.. because `string-width` lib was the topic of an error presented above.

I still don't know what could have gone wrong and what actually caused the error. Anyway, it's gone now and I think we can close the issue 🤷🏼‍♂️.
